### PR TITLE
dbeaver/pro#2622 Add semantic error problem markers

### PR DESCRIPTION
**Pull request by E1izabeth**
Now we show problem markers for semantic errors
image

We have multiple messages in one tooltip and can jumpby to the issue source from marker tooltip or from Problems view by clicking
image

### CODE REVIEW

**Review by serjiokov — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLSemanticErrorAnnotation.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import org.eclipse.swt.widgets.Canvas;
+import org.eclipse.ui.texteditor.MarkerAnnotation;
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.model.DBIcon;
+import org.jkiss.dbeaver.model.sql.semantics.SQLQueryRecognitionProblemInfo;
+import org.jkiss.dbeaver.ui.DBeaverIcons;
+
+public class SQLSemanticErrorAnnotation extends MarkerAnnotation implements IAnnotationPresentation {
+    public static final String MARKER_TYPE = "org.jkiss.dbeaver.ui.editors.sql.semanticProblemMarker";
+    public static final String TYPE = "org.eclipse.ui.workbench.texteditor.error";
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLSemanticErrorAnnotation.java]: Already have this constant in SQLProblemAnnotation

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLEditorSemanticMarkersManager.java [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    };
+
+    @NotNull
+    private final SQLEditorBase editor;
+    @Nullable
+    private volatile SQLDocumentSyntaxContext syntaxContext;
+    @NotNull
+    private final Object syncRoot = new Object();
+    @NotNull
+    private final HashMap<SQLDocumentScriptItemSyntaxContext, UpdateMarkersOperationInfo> queuedOperations = new HashMap<>();
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLEditorSemanticMarkersManager.java]: Use Map

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Review by HocKu7 — APPROVED** [review]

**Comment by Matvey16** [comment]
This seems to not work for me. Can you try downloading jenkins build and trying this case?
1. Open new script
2. input `select * from sdsdgsggsdfgsdfgsdhgh`
3. Hover the cursor on exclamation mark

Nothing appears on screen after that and everytime i do it there is an error in the log

```
java.lang.IllegalArgumentException: Index out of bounds
	at org.eclipse.swt.SWT.error(SWT.java:4903)
	at org.eclipse.swt.SWT.error(SWT.java:4837)
	at org.eclipse.swt.SWT.error(SWT.java:4808)
	at org.eclipse.swt.custom.StyledText.getTextBounds(StyledText.java:4887)
	at org.jkiss.dbeaver.ui.editors.sql.util.SQLAnnotationHover$LinkListInformationControl.setVisible(SQLAnnotationHover.java:274)
	at org.eclipse.jface.text.AbstractInformationControlManager.showInformationControl(AbstractInformationControlManager.java:1240)
	at org.eclipse.jface.text.source.AnnotationBarHoverManager.showInformationControl(AnnotationBarHoverManager.java:353)
	at org.eclipse.jface.text.AbstractInformationControlManager.internalShowInformationControl(AbstractInformationControlManager.java:1191)
	at org.eclipse.jface.text.AbstractInformationControlManager.presentInformation(AbstractInformationControlManager.java:1120)
	at org.eclipse.jface.text.AbstractHoverInformationControlManager.presentInformation(AbstractHoverInformationControlManager.java:884)
	at org.eclipse.jface.text.AbstractInformationControlManager.setInformation(AbstractInformationControlManager.java:431)
	at org.eclipse.jface.text.source.AnnotationBarHoverManager.computeInformation(AnnotationBarHoverManager.java:340)
	at org.eclipse.jface.text.AbstractInformationControlManager.doShowInformation(AbstractInformationControlManager.java:1101)
	at org.eclipse.jface.text.AbstractHoverInformationControlManager$MouseTracker.mouseHover(AbstractHoverInformationControlManager.java:511)
	at org.eclipse.swt.widgets.TypedListener.handleEvent(TypedListener.java:225)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:91)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4285)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1160)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4083)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3673)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1151)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1042)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:152)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:639)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:546)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:173)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:291)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.invokeFramework(DBeaverLauncher.java:648)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.basicRun(DBeaverLauncher.java:608)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.run(DBeaverLauncher.java:1459)
```


https://github.com/user-attachments/assets/87b97dc3-16e5-4d8c-9e8e-910113b1ea15

**Comment by Matvey16** [comment]
Verified

**Pull request merged**